MaterialDesignThemes.Wpf/Stepper.cs

```diff
@@ -0,0 +1,352 @@
+        {
+            args.CanExecute = true;
+        }
+
+        private void ContinueHandler(object sender, ExecutedRoutedEventArgs args)
+        {
+            StepperNavigationEventArgs navigationArgs = new StepperNavigationEventArgs(ContinueNavigationEvent, this, _controller.ActiveStep, _controller.NextStep, false);
+            RaiseEvent(navigationArgs);
+
+            if (!navigationArgs.Cancel)
+            {
+                _controller.Continue();
+            }
+        }
+
+        private void CanExecuteStepSelectedHandler(object sender, CanExecuteRoutedEventArgs args)
+        {
+            args.CanExecute = !IsLinear;
+        }
+
+        private void StepSelectedHandler(object sender, ExecutedRoutedEventArgs args)
+        {
+            if (!IsLinear)
+            {
+                StepperNavigationEventArgs navigationArgs = new StepperNavigationEventArgs(StepNavigationEvent, this, _controller.ActiveStep, ((StepperStepViewModel)args.Parameter).Step, false);
+                RaiseEvent(navigationArgs);
+
+                if (!navigationArgs.Cancel)
+                {
+                    _controller.GotoStep((StepperStepViewModel)args.Parameter);
+                }
+            }
+        }
+
+        private void PropertyChangedHandler(object sender, PropertyChangedEventArgs args)
+        {
+            if (sender == _controller && args.PropertyName == nameof(_controller.ActiveStepContent)
+                    && _controller.ActiveStepContent != null && Layout == StepperLayout.Horizontal)
+            {
+                // there is no event raised if the Content of a ContentControl changes
+                //     therefore trigger the animation in code
+                PlayHorizontalContentAnimation();
+            }
+        }
+
+        private void PlayHorizontalContentAnimation()
+        {
+            // there is no event raised if the Content of a ContentControl changes
+            //     therefore trigger the animation in code
+            if (Layout == StepperLayout.Horizontal)
+            {
+                Storyboard storyboard = (Storyboard)TryFindResource("horizontalContentChangedStoryboard");
+                FrameworkElement element = GetTemplateChild("PART_horizontalContent") as FrameworkElement;
+
+                if (storyboard != null && element != null)
+                {
+                    storyboard.Begin(element);
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// The layout of a <see cref="Stepper"/>.
+    /// </summary>
+    public enum StepperLayout : byte
+    {
+        Horizontal,
+        Vertical
+    }
+
+    /// <summary>
+    /// The argument for the <see cref="Stepper.StepValidation"/> event and the <see cref="Stepper.StepValidationCommand"/> command.
+    /// It holds the <see cref="IStep"/> to validate.
+    /// </summary>
+    public class StepValidationEventArgs : RoutedEventArgs
```

Comment:
Validation has been removed.

Reply:
@CookiesAndCoffee I recently decided to start a new project for not specified or scope-crashing controls. Have a look at [Material Design Extensions](https://github.com/spiegelp/MaterialDesignExtensions). It contains the old `Stepper` before the refactoring.

Reply:
@spiegelp This is a great solution while this pull request is work in progress.